src/isValid.js

```diff
@@ -1,8 +1,7 @@
-'use strict';
-import curry from 'lodash.curry';
+import curry from 'lodash.curry'
 
 export default curry((date) => {
-    if(Object.prototype.toString.call(date) !== "[object Date]") return false;
-    let unixTime = date.getTime();
-    return unixTime === unixTime;
-});
+  if (Object.prototype.toString.call(date) !== "[object Date]") return false
+
+  return !isNaN(date.getTime())
```

Comment:
I specifically didn't use the global `isNaN` function because it does implicit type coercion, which can result in false negatives. I didn't use the new non-broken version `Number.isNaN` because it's only available on new versions of Node and some browsers.